admin/ajax/class-recalculate-scores-ajax.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * @package WPSEO\Admin|Ajax
+ */
+
+/**
+ * Class WPSEO_Recalculate_Scores
+ *
+ * This class handles the SEO score recalculation for all posts with a filled focus keyword
+ */
+class WPSEO_Recalculate_Scores_Ajax {
+
+	/**
+	 * @var int
+	 */
+	private $posts_per_page = 20;
+
+	/**
+	 * @var array The fields which should be always queries, can be extended by array_merge
+	 */
+	private $query_fields   = array(
+		'post_type'      => 'any',
+		'meta_key'       => '_yoast_wpseo_focuskw',
+	);
+
+	/**
+	 * @var array The options stored in the database
+	 */
+	private $options;
+
+	/**
+	 * Initialize the AJAX hooks
+	 */
+	public function __construct() {
+		add_action( 'wp_ajax_wpseo_recalculate_scores', array( $this, 'recalculate_scores' ) );
+		add_action( 'wp_ajax_wpseo_update_score', array( $this, 'save_score' ) );
+	}
+
+	/**
+	 * Start recalculation
+	 */
+	public function recalculate_scores() {
+		if ( $this->verify_nonce() ) {
```

Comment:
Why don't you use `check_ajax_referrer` and why do you need a wrapper function for this?